### DISCUSSION TITLE
How to programatically navigate from one page to another?

### DISCUSSION THREAD

**Discussion by gmgallo** [microsoft/TemplateStudio#4724] [Q&A]
In a MVVM template project I'm trying to navigate from one page to another programmatically, for example when the user must update a setting in another page before continuing. The idea behind is that there is only one place where settings must be changed and saved.
I know that NavigationService  and NavigationHelper are the way to go, but I can't figure out the right way to use them in code.

**Comment by tinmac** [comment]
To navigate from Pae to Page in code you would do something like this

`this.Frame.Navigate(typeof(Main_Page));`

The above example resides in a Login Page code behind
